src/main/scala/elevate/rise/rules/algorithmic.scala

```diff
@@ -55,24 +56,40 @@ object algorithmic {
         Success((app(map, lambda(untyped(x), gx)) >> map(f)) :: e.t)
       case _ => Failure(mapLastFission)
     }
-    override def toString = s"mapLastFission"
+    override def toString: String = s"mapLastFission"
   }
 
   // identities
 
   def idAfter: Strategy[Rise] = ` -> id`
   case object ` -> id` extends Strategy[Rise] {
     def apply(e: Rise): RewriteResult[Rise] = Success((typed(e) |> id) :: e.t)
-    override def toString = "idAfter"
+    override def toString: String = "idAfter"
   }
 
   def liftId: Strategy[Rise] = `id -> *id`
   case object `id -> *id` extends Strategy[Rise] {
     def apply(e: Rise): RewriteResult[Rise] = e match {
-      case App(Id(), arg) => Success(app(map(id), arg) :: e.t)
+      case App(i@Id(), arg) => i.t match {
+        case FunType(ArrayType(_,_),_) =>
+          Success(LCNF(app(map(id), arg) :: e.t))
+        case _ => Failure(liftId)
+      }
       case _              => Failure(liftId)
     }
-    override def toString = "liftId"
+    override def toString: String = "liftId"
+  }
+
+  def idToCopy: Strategy[Rise] = `id -> fun(x => x)`
```

Comment:
I am not a fan of the name. For me this goes from the id primitive to the id function. Copies make me think about memory.